sympy/physics/control/lti.py

```diff
@@ -0,0 +1,1033 @@
+from sympy import (
+    Symbol, simplify, oo, exp,
+    Poly, lcm, LC, degree, Integral, integrate,
+    Matrix, eye, zeros, Basic, latex,
+    ShapeError, ImmutableMatrix, MutableMatrix,
+    SparseMatrix, MutableDenseMatrix
+)
+
+from sympy.printing import sstr
+
+# import mpmath for numercial results
+from mpmath import expm, quad, matrix as mpm_matrix
+
+__all__ = ['StateSpaceModel', 'TransferFunctionModel']
+
+_matrixTypes = (
+    Matrix, ImmutableMatrix, MutableMatrix, SparseMatrix, MutableDenseMatrix)
+
+
+class StateSpaceModel(Basic):
+    """State space model (ssm) of a linear, time invariant control system
+
+    Represents the standard state-space model with state matrix A, input matrix B, output matrix C, and
+    transmission matrix D. This makes the linear control system:
+        (1) x'(t) = A * x(t) + B * u(t);    x in R^n , u in R^k
+        (2) y(t)  = C * x(t) + D * u(t);    y in R^m
+    where u(t)  is any input signal, y(t) the corresponding output, and x(t) the system's state.
+
+    Parameters
+    ==========
+
+    *arg : TransferFunctionModel, Sympy-Matrices
+        tfm to construct the state space model from, or the Matrices A, B, C, D as StateSpaceModel(A, B, C, D)
+
+    Examples
+    ========
+
+    >>> from sympy import Matrix, Symbol
+    >>> from sympy.physics.control import StateSpaceModel, TransferFunctionModel
+
+    The easiest way to create a StateSpaceModel is via four matrices:
+
+    >>> A, B, C, D = Matrix([1,2]), Matrix([2,3]), Matrix([2]), Matrix([0])
+    >>> StateSpaceModel(A, B, C, D)
+    StateSpaceModel(
+    Matrix([
+    [1],
+    [2]]),
+    Matrix([
+    [2],
+    [3]]),
+    Matrix([[2]]),
+    Matrix([[0]]))
+
+    One can use less matrices as well. The rest will be filled with a minimum of
+    zeros:
+
+    >>> StateSpaceModel(A, B)
+    StateSpaceModel(
+    Matrix([
+    [1],
+    [2]]),
+    Matrix([
+    [2],
+    [3]]),
+    Matrix([[0]]),
+    Matrix([[0]]))
+
+    One can also use a TansferFunctionModel to create a StateSpaceModel
+
+    >>> s = Symbol('s')
+    >>> tfm = TransferFunctionModel(Matrix([1/s, s/(1 + s**2)]))
+    >>> StateSpaceModel(tfm)
+    StateSpaceModel(
+    Matrix([
+    [0, -1, 0],
+    [1,  0, 0],
+    [0,  1, 0]]),
+    Matrix([
+    [1],
+    [0],
+    [0]]),
+    Matrix([
+    [1, 0, 1],
+    [1, 0, 0]]),
+    Matrix([
+    [0],
+    [0]]))
+
+    See Also
+    ========
+
+    TransferFunctionModel: transfer function model of a lti system
+
+    References
+    ==========
+
+    Joao P. Hespanha, Linear Systems Theory. 2009.
+    """
+
+    def __init__(self, *arg):
```

Comment:
We try to avoid using `*arg, **kwarg`, it's better to be explicit. See comments above.

Reply:
yes, I'm trying to define the constructor with `__new__(cls, args)` instead of `__init__`. StateSpaceModel will use `Basic.__new__(cls, A, B, C, D, s)`.